src/corva/service/cache_sdk.py

```diff
@@ -56,19 +56,29 @@ class UserRedisSdk:
 
     SIXTY_DAYS: int = int(datetime.timedelta(days=60).total_seconds())
 
-    def __init__(self, hash_name: str, redis_dsn: str, use_fakes: bool = False):
+    def __init__(
+        self,
+        hash_name: str,
+        redis_dsn: str,
+        use_fakes: bool = False,
+        redis_client=None,
+    ):
         use_lua_52 = False
+        # use either provided redis client, or initialize "fake" client
+        # (usually used for tests), or initialize real new client
         if use_fakes:
-            client = fakeredis.FakeRedis.from_url(url=redis_dsn, decode_responses=True)
+            redis_client = fakeredis.FakeRedis.from_url(
+                url=redis_dsn, decode_responses=True
+            )
             use_lua_52 = True
-        else:
-            client = redis.Redis.from_url(url=redis_dsn, decode_responses=True)
+        elif redis_client is None:
+            redis_client = redis.Redis.from_url(url=redis_dsn, decode_responses=True)
```

Comment:
This functionality might be obsolete now, since we always pass `redis_client` in our handlers

If backward-compatibility was a concern, when designing this - this object always gets initialized by the sdk, so users are not supposed to init it themselves